common/state-machine-lib/src/main/java/org/eclipse/dataspaceconnector/common/statemachine/StateMachine.java

```diff
@@ -94,42 +104,50 @@ public boolean isActive() {
 
     private Runnable loop() {
         return () -> {
-            while (active.get()) {
-                try {
-                    var processed = processors.stream()
-                            .mapToLong(StateProcessor::process)
-                            .sum();
-
-                    if (processed == 0) {
-                        Thread.sleep(waitStrategy.waitForMillis());
-                    }
-                    waitStrategy.success();
-                } catch (Error | InterruptedException e) {
-                    active.set(false);
-                    monitor.severe(format("StateMachine [%s] unrecoverable error", name), e);
-                } catch (Throwable e) {
-                    try {
-                        monitor.severe(format("StateMachine [%s] error caught", name), e);
-                        Thread.sleep(waitStrategy.retryInMillis());
-                    } catch (InterruptedException ex) {
-                        active.set(false);
-                        monitor.severe(format("StateMachine [%s] unrecoverable error", name), e);
-                    }
-                }
+            if (!active.get()) {
```

Comment:
Since the same condition is evaluated some lines below, it will be more immediate to avoid the return:
```
            if (active.get()) {
                long delay = performLogic();
                submit(delay);
            }
```

Reply:
That double checking is intentional. If the machine is set to inactive while performLogic() is running, we don't want to schedule a next execution for nothing. Does that make sense?

Reply:
The next iteration would exit as soon as it starts, not a big deal, considering that active is set to false once in the state machine lifetime: at shutdown. 
I don't want to talk about performances, but considering how many times the loop will call this method, one less evaluation could be good overall.

Reply:
Implemented